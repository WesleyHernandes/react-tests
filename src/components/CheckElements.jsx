export default function Menu() {
  return (
    <nav>
      <a className="links" role="link">
        Link 01
      </a>
      <a className="links" role="link">
        Link 02
      </a>
      <a className="links" role="link">
        Link 03
      </a>
      <a className="links" role="link">
        Link 04
      </a>
    </nav>
  );
}
