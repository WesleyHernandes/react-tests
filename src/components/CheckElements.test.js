import { render, screen } from "@testing-library/react";
import CheckElements from "./CheckElements";

test("Deve renderizar um link na página", () => {
  render(<CheckElements />);
  const link1 = screen.getByText("Link 01");
  expect(link1).toBeInTheDocument();
});

test("Deve renderizar uma lista de links", () => {
  render(<CheckElements />);
  const links = screen.getAllByRole("link");
  expect(links).toHaveLength(4);
});

test("Não deve renderizar o link 05", () => {
  render(<CheckElements />);
  const link5 = screen.queryByText("Link 05");
  expect(link5).not.toBeInTheDocument();
});

test("Deve renderizar uma lista de links com a classe link", () => {
  render(<CheckElements />);
  const links = screen.getAllByRole("link");
  links.forEach((link) => expect(link).toHaveClass("links"));
  expect(links).toMatchSnapshot();
});
