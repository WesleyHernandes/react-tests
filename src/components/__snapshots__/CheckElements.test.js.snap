// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma lista de links com a classe link 1`] = `
Array [
  <a
    class="links"
    role="link"
  >
    Link 01
  </a>,
  <a
    class="links"
    role="link"
  >
    Link 02
  </a>,
  <a
    class="links"
    role="link"
  >
    Link 03
  </a>,
  <a
    class="links"
    role="link"
  >
    Link 04
  </a>,
]
`;
